cql3/selection/selection.cc

```diff
@@ -395,14 +395,17 @@ bool result_set_builder::restrictions_filter::do_filter(const selection& selecti
     }
 
     auto static_row_iterator = static_row.iterator();
-    auto row_iterator = row.iterator();
+    auto row_iterator = row ? std::optional<query::result_row_view::iterator_type>(row->iterator()) : std::nullopt;
```

Comment:
I'm surprised this syntax worked - the alternatives don't have the same type...

Reply:
The ternary operator assumes it will return the type of the first alternative, and if the second one is implicitly convertible to that type as well, it will work fine. (that's why I explicitly specified the type in the first alternative instead of:
```cpp
std::optional<query::result_row_view::iterator_type> row_iterator = row ? row->iterator() : std::nullopt;
```
the one above wouldn't work.